src/plugins/data_view_management/public/management_app/data_view_management_service.ts

```diff
@@ -0,0 +1,293 @@
+
+  private internalState$: BehaviorSubject<DataViewMgmtState>;
+  state$: BehaviorObservable<DataViewMgmtState>;
+
+  private updateState = (newState: Partial<DataViewMgmtState>) => {
+    this.internalState$.next({ ...this.state$.getValue(), ...newState });
+  };
+
+  private getKbnUrl = (dataViewId: string) =>
+    setStateToKbnUrl(
+      APP_STATE_STORAGE_KEY,
+      {
+        fieldTypes: ['conflict'],
+        tab: 'indexedFields',
+      },
+      { useHash: this.services.uiSettings.get('state:storeInSessionStorage') },
+      this.services.application.getUrlForApp('management', {
+        path: `/kibana/dataViews/dataView/${encodeURIComponent(dataViewId)}`,
+      })
+    );
+
+  private getTags = async (dataView: DataView) => {
+    if (dataView) {
+      const defaultIndex = await this.services.uiSettings.get('defaultIndex');
+      const tags = getTags(
+        dataView,
+        dataView.id === defaultIndex,
+        this.services.dataViews.getRollupsEnabled()
+      );
+
+      return tags;
+    }
+    return [];
+  };
+
+  async updateScriptedFields() {
+    const dataView = this.state$.getValue().dataView;
+    if (dataView) {
+      const scriptedFieldRecords = dataView.getScriptedFields();
+      const scriptedFields = Object.values(scriptedFieldRecords);
+
+      const scriptedFieldLangs = Array.from(
+        scriptedFields.reduce((acc: Set<string>, field) => {
+          if (field.lang) {
+            acc.add(field.lang);
+          }
+          return acc;
+        }, new Set<string>())
+      );
+
+      this.updateState({
+        scriptedFields,
+        scriptedFieldLangs: convertToEuiFilterOptions(scriptedFieldLangs),
+      });
+    }
+  }
+
+  async setDataView(dataView: DataView) {
+    this.updateState({ isRefreshing: true });
+
+    const fieldRecords = dataView.fields
+      .filter((field) => !field.scripted)
+      .reduce((acc, field) => {
+        acc[field.name] = field;
+        return acc;
+      }, {} as Record<string, DataViewField>);
+
+    const fields = Object.values(fieldRecords);
+    const indexedFieldTypes: string[] = [];
+    fields.forEach((field) => {
+      // for conflicted fields, add conflict as a type
+      if (field.type === 'conflict') {
+        indexedFieldTypes.push('conflict');
+      }
+      if (field.esTypes) {
+        // add all types, may be multiple
+        field.esTypes.forEach((item) => indexedFieldTypes.push(item));
+      }
+    });
+
+    const allowedAsString = (await this.allowedTypes).map((item) => item.name);
+
+    this.services.savedObjectsManagement
+      .getRelationships(DATA_VIEW_SAVED_OBJECT_TYPE, dataView.id!, allowedAsString)
+      .then((resp) => {
+        this.updateState({
+          relationships: resp.relations.map((r) => ({ ...r, title: r.meta.title! })),
+        });
+      });
+
+    this.updateState({
+      dataView,
+      fields,
+      indexedFieldTypes,
+      fieldConflictCount: fields.filter((field) => field.type === 'conflict').length,
```

Comment:
Can scripted fields cause a conflict? 

```suggestion
      fieldConflictCount: dataView.fields.filter((field) => field.type === 'conflict').length,
```

Reply:
This is purposeful as the DataViewLazy field list is a separate object and this code aims to minimize changes from DataView to DataViewLazy

Reply:
`fields` excludes scripted fields when defined in `setDataView`. So my question was whether scripted fields still need to be checked for conflicts.

I tested it and it seems like scripted fields "override" normal fields hence we don't report on type conflicts.

Reply:
That's correct, scripted fields override, no conflict possible.